draft-sheffer-oauth-rfc8725bis.md

```diff
@@ -276,16 +283,14 @@ attacker gets hold of such a token  {{Langkemper}}.
 
 
 
-## Incorrect Composition of Encryption and Signature {#incorrect-composition-of-encryption-and-signature}
+## Incorrect Use and Composition of Encryption and Signature {#incorrect-composition-of-encryption-and-signature}
 
+Most authentication use cases only require a simple signed JWT as their token. However verifiers don't always check that the received JWT is a signed JWS as opposed to an encrypted JWE structure. This can result in vulnerabilities, in particular when the verifier's asymmetric key includes the private key, and when the verifier's library does not distinguish between successful decryption and successful signature validation {{CVE-2023-51774}}.
```

Comment:
I don't understand what the phrase "when the verifier's asymmetric key includes the private key" is getting at.  How does this relate to the vulnerability?

Reply:
On the receiver side, if all you want is a JWS then you only need a public key in the JWK, to verify the signature.

But if you're accepting a JWE, you also need a private key in the JWK to decrypt the thing. So the attack (sending a JWE instead of a JWS) only works if there's a private key on the receiver's side.